simson/common/data_transformations.py

```diff
@@ -71,23 +74,47 @@ def gdp_regression(historic_stocks_pc, gdppc, prediction_out):
     for i_region in range(shape_out[1]):
         for i_good in range(shape_out[2]):
 
-            def fitting_function(prms):
+            def sigmoid_fitting_function(prms):
                 return (
                     prms[0] / (1. + np.exp(prms[1]/gdppc[:n_historic,i_region]))
                     ) - historic_stocks_pc[:,i_region,i_good]
                 # Lagrangian multiplier to ensure matching last point:
                 # + prms[2] * prms[0] / (1. + np.exp(prms[1]/gdppc[i_lh,0])) - stocks_pc[-1,0,i] )
-            prms_out = least_squares(
-                fitting_function,
-                x0=np.array([2.*gdppc[i_lh,i_region], historic_stocks_pc[-1,i_region,i_good]]),
-                gtol=1.e-12
-            )
-            assert prms_out.success
 
-            pure_prediction[:,i_region,i_good] = prms_out.x[0] / (
-                1. + np.exp(prms_out.x[1]/gdppc[:,i_region])
+            def exponential_fitting_function(prms):
+                return (
+                    prms[0] * (1 - np.exp(-prms[1]*gdppc[:n_historic,i_region]))
+                ) - historic_stocks_pc[:,i_region,i_good]
+
+            if fitting_function_type == 'sigmoid':
+                prms_out = least_squares(
+                    sigmoid_fitting_function,
+                    x0=np.array([2.*gdppc[i_lh,i_region], historic_stocks_pc[-1,i_region,i_good]]),
+                    gtol=1.e-12
+                )
+
+                pure_prediction[:, i_region, i_good] = prms_out.x[0] / (
+                        1. + np.exp(prms_out.x[1] / gdppc[:, i_region])
+                )
+            elif fitting_function_type == 'exponential':
+                prms_out = least_squares(
+                    exponential_fitting_function,
+                    x0=np.array([400, 1]),
```

Comment:
No, I think they should be different, at least the second one (the first one should be the saturation level which I guess should be similar with both approaches, but the way it is calculated in the sigmoid case is I think specific to the plastics model [I guess it might make sense to change that structure at some point as this is in the common section now - e.g. by specifying in the config whether and how the initial guess of saturation level and other parameters should be calculated?]). 

In the former steel model, we had different version of this regression, one, where the parameters where optimised like here and one where we calculated it numerically. As I changed the inflow driven model calculation, I steel need to look at the numbers to make this decision so this might change soon again.

Reply:
Okay, I didn't release this was a model specific initial guess. If you were using both options (exponential and sigmoid) for the steel model (e.g. to compare results from different approaches), you would then have the same x0, right? Or is it also dependent on the fitting function? Maybe, we could pass x0 as a parameter to the gdp_regression function, but I think I would keep hard-coded values somewhere (and not require the user to specify it in the config), maybe directly in the steel and plastic models, or putting (or calling) this gdp_regression in the steel / plastic directories rather than common. For now, please just add a little comment to the code with your explanation above and I will think what to do with it later!

Reply:
The initial guess x0 would be different for the different fitting functions, at least for the second parameter. In both fitting functions here the first parameter describes the saturation level, whilst the second one defines some other valuable that influences the curvature of the function which will be very different depending on the form of the function. One could also imagine fitting functions which don't use a saturation level at all in the future, hence they should always be different. I will add the comment like you said and as said I also need to look at the numbers a little more myself - am happy to change the location of the x0 values with whatever you propose!